dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+            .withProjectDir(testProjectDir)
+            .withArguments(taskName)
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @ParameterizedTest
+    @MethodSource("uploadTasks")
+    fun `M success W upload { checkProjectDependecies set to warn }`(taskName: String) {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_set_to_warn.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(taskName)
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M not contain any uploadTasks W minifyNotEnabeld`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_minify_not_enabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("tasks")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        val uploadTask = result.output
+            .split("\n")
+            .firstOrNull { it.startsWith("upload") }
+        assertThat(uploadTask).isNull()
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun stubFile(destination: File, content: String) {
+        with(destination.outputStream()) {
+            write(content.toByteArray())
+        }
+    }
+
+    private fun stubGradleBuildFromResourceFile(resourceFilePath: String) {
+        javaClass.classLoader.getResource(resourceFilePath)?.file?.let {
+            File(it).copyTo(buildFile)
+        }
+    }
+
+    private fun getTestConfigurationClasspath(): List<File> {
+        // we will use this classpath for the GradleRunner to make sure we have all the
+        // required classes (AndroidGradle and KotlinGradle) to build the plugin. GradleRunner
+        // creates its own process and by default the classpath used is the one from the `main`
+        // configuration.
+        val properties = Properties()
+        properties["implementation-classpath"] = System.getProperty("java.class.path")
+        return PluginUnderTestMetadataReading
+            .readImplementationClasspath("gradle-runner-classpath", properties)
+    }
+
+    // endregion
+
+    companion object {
+        val ANDROID_MANIFEST_FILE_CONTENT = """
+            <?xml version="1.0" encoding="utf-8"?>
+            <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+                      package="com.example.variants">
+
+                <application
+                    android:allowBackup="true"
+                    android:supportsRtl="true">
```

Comment:
let's add also `exported=true` here, so that compilation succeeds for API 31 (I will handle the rest in the https://github.com/DataDog/dd-sdk-android-gradle-plugin/pull/53)